fdbserver/workloads/TenantManagement.actor.cpp

```diff
@@ -560,19 +560,103 @@ struct TenantManagementWorkload : TestWorkload {
 		}
 	}
 
+	ACTOR Future<Void> renameTenant(Database cx, TenantManagementWorkload* self) {
+		// Currently only supporting MANAGEMENT_DATABASE op, so numTenants should always be 1
+		// state OperationType operationType = TenantManagementWorkload::randomOperationType();
+		int numTenants = 1;
+
+		state std::vector<TenantName> oldTenantNames;
+		state std::vector<TenantName> newTenantNames;
+		// For the future in supporting different operation types and number of tenants:
+		// The number of tenants we rename should be min(numTenants, self->createdTenants.size())
+		state bool tenantExists = false;
+		state bool tenantNotFound = false;
+		for (int i = 0; i < numTenants; ++i) {
+			TenantName oldTenant = self->chooseTenantName(false);
+			TenantName newTenant = self->chooseTenantName(false);
+			newTenantNames.push_back(newTenant);
+			oldTenantNames.push_back(oldTenant);
+			if (!self->createdTenants.count(oldTenant)) {
+				tenantNotFound = true;
+			}
+			if (self->createdTenants.count(newTenant)) {
+				tenantExists = true;
+			}
+		}
+
+		loop {
+			try {
+				ASSERT(oldTenantNames.size() == 1);
+				state int tenantIndex = 0;
+				for (; tenantIndex != oldTenantNames.size(); ++tenantIndex) {
+					state TenantName oldTenantName = oldTenantNames[tenantIndex];
+					state TenantName newTenantName = newTenantNames[tenantIndex];
+					// Perform rename, then check against the DB for the new results
+					wait(ManagementAPI::renameTenant(cx.getReference(), oldTenantName, newTenantName));
```

Comment:
Let's add an assertion after this wait statement that `!tenantNotFound && !tenantExists`. If either of those is true, then we were expecting the operation to fail.